crates/project_model/src/lib.rs

```diff
@@ -365,58 +396,96 @@ impl ProjectWorkspace {
                     }
                 }
             }
-            ProjectWorkspace::Cargo { cargo, sysroot } => {
+            ProjectWorkspace::Cargo { cargo, sysroot, rustc } => {
                 let (public_deps, libproc_macro) =
                     sysroot_to_crate_graph(&mut crate_graph, sysroot, target, load);
 
                 let mut cfg_options = CfgOptions::default();
                 cfg_options.extend(get_rustc_cfg_options(target));
 
                 let mut pkg_to_lib_crate = FxHashMap::default();
-                let mut pkg_crates = FxHashMap::default();
 
                 // Add test cfg for non-sysroot crates
                 cfg_options.insert_atom("test".into());
                 cfg_options.insert_atom("debug_assertions".into());
 
+                let mut rustc_pkg_crates = FxHashMap::default();
```

Comment:
unfortunately, this match arm becomes more complex since it must load `rustc_workspace` (if provided) and `cargo`. I tried to clean it up a bit by pulling out the logic around adding a `pkg`, `tgt` crate root to a separate function, but there is still quite a bit of repitition.

Reply:
Yeah, I'll look into fixing this after the PR is meged. I bet the code could be refactored somewhat more radically.

Reply:
yea, don't feel confident enough with the codebase to go ahead and do that, but I can definitely some possible refactorings.